src/main/java/com/google/sps/servlets/GetUsersServlet.java

```diff
@@ -0,0 +1,127 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Arrays;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
+import com.google.gson.Gson;
+import com.google.sps.data.FormattedUser;
+import com.google.sps.data.Project;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import com.google.cloud.datastore.DatastoreException;
+		
+@WebServlet("/get-users")
+public class GetUsersServlet extends HttpServlet {
+
+	@Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response) 
+			throws IOException {
+    		 // Allow CORS so frontend can access it			
+		 response.addHeader("Access-Control-Allow-Origin", "*");
+		 response.addHeader("Access-Control-Allow-Headers", 
+				"Origin, X-Requested-With, Content-Type, Accept, Authorization");
+		 response.addHeader("Access-Control-Allow-Credentials", "true");
+		 response.addHeader("Access-Control-Allow-Methods", "GET,POST,PUT,DELETE,OPTIONS,HEAD");
+		 
+		 // Get the username from user
+		 String username = Jsoup.clean(request.getParameter("username"), Whitelist.none());
+
+		 Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+		 FormattedUser[] users = loadUsers(username, datastore);
+		
+		 Gson gson = new Gson();	
+
+		 // Return project objects to frontend
+		 response.setContentType("application/json");
+		 response.getWriter().println(gson.toJson(users));
+	}
+
+	// Get users associated with user from Datastore
+	private FormattedUser[] loadUsers(String username, Datastore datastore) 
+			throws DatastoreException {
+		 // Initialize Array list for usernames of project partners	
+		 List<String> allUsers = new ArrayList<>();
```

Comment:
Why not just initialize this as a set and add the users to it directly? I don't see allUsers being used anywhere else besides being copied to the set